junit-platform-console/src/main/java/org/junit/platform/console/ConsoleLauncherExecutionResult.java

```diff
@@ -38,7 +39,10 @@
 	 */
 	private static final int FAILED = -1;
 
-	public static int computeExitCode(TestExecutionSummary summary) {
+	public static int computeExitCode(TestExecutionSummary summary, CommandLineOptions options) {
+		if (options.isFailIfNoTests() && summary.getTestsFoundCount() == 0) {
+			return FAILED;
```

Comment:
I think we should introduce a separate exit code, e.g. `2` for `NO_TESTS_FOUND`.